store-api-server/src/payment/service/payment.service.ts

```diff
@@ -868,6 +807,123 @@ WHERE provider IN ('tezpay', 'wert')
+            {
+              headers: {
+                "Authorization": `ApiKey ${SIMPLEX_API_KEY}`
+              }
+            }
+        );
+        return eventsResponse.data;
+      } catch (error) {
+        let errorMessage;
+        if (error instanceof Error) {
+          if (axios.isAxiosError(error) && error.response) {
+            errorMessage = error.response?.data?.error || error.response?.data;
+            let errors = error.response?.data?.errors;
+            if (errors && typeof errors == "object") {
+              errorMessage = error.response?.data?.error + "---DETAILS:---" + JSON.stringify(errors);
+            }
+            Logger.warn("getSimplexEvents ERROR" + errorMessage);
+            throw new Error(`there is problem simplex api getSimplexEvents please contact your backend services`);
+          } else {
+            Logger.warn("Unexpected error simplex api getSimplexEvents instance of error", error.message);
+          }
+        } else {
+          Logger.warn("Unexpected error simplex api getSimplexEvents");
+        }
+        throw new Error(`there is problem simplex api getSimplexEvents please contact your backend services`);
+      }
+    }
+
+    async function deleteSimplexEvents(eventId: string) {
+      try {
+        let eventsResponse = await axios.delete(
+            SIMPLEX_API_URL + "/wallet/merchant/v2/events/" + eventId,
+            {
+              headers: {
+                "Authorization": `ApiKey ${SIMPLEX_API_KEY}`
+              }
+            }
+        );
+        return eventsResponse.data;
+      } catch (error) {
+        let errorMessage;
+        if (error instanceof Error) {
+          if (axios.isAxiosError(error) && error.response) {
+            errorMessage = error.response?.data?.error || error.response?.data;
+            let errors = error.response?.data?.errors;
+            if (errors && typeof errors == "object") {
+              errorMessage = error.response?.data?.error + "---DETAILS:---" + JSON.stringify(errors);
+            }
+            Logger.warn("getSimplexEvents ERROR" + errorMessage);
+          } else {
+            Logger.warn("Unexpected error simplex api deleteSimplexEvents instance of error", error.message);
+          }
+        } else {
+          Logger.warn("Unexpected error simplex api deleteSimplexEvents");
+        }
+      }
+    }
+
+    function getSimplexPaymentStatus(event: { name: any; }) {
+      let paymentStatus: PaymentStatus;
+      switch (event?.name) {
+        case "payment_simplexcc_approved":
+          paymentStatus = PaymentStatus.SUCCEEDED;
+          break;
+        case "payment_simplexcc_declined":
+          paymentStatus = PaymentStatus.FAILED;
+          break;
+        default:
+          Logger.error(`Unhandled payment status ${event?.name}`);
+          throw Err("Unknown simplex events");
+      }
+      return paymentStatus;
+    }
+
+    let eventsResponse = await getSimplexEvents();
```

Comment:
Is there no way to call them to get events for specific payment ids? Because this thing here will just grow and grow right? Always giving us the complete set of events since the beginning?

We have to find a way to request just the events that we care about.

Edit: ah I see that below you cleanup processed events with the delete call. But is this really the way we are supposed to use Simplex? It seems a bit odd to _delete_ events - because that sounds to me like it also erases important records. I may be wrong here but if you're not sure either about this best to have a look at their API docs/ask their team to get some clarification about whether we are using delete for wrong reasons here.

Reply:
ahh you are definitely right. I pushed to want to with query param paymentid or userid. However, they answered did not do this. We have to get all events. That's crazy! I sent message twice via telegram :( 
By the way they ask to delete events odd. :/

Reply:
Ok well if this is what they recommended, then lets go forward like this.

We still have our own book keeping, it just feels so wrong to delete records at the payment provider.